File: src/main/java/com/kishlaly/ta/model/indicators/EMA.java
package com.kishlaly.ta.model.indicators;

import com.kishlaly.ta.model.EntityWithDate;

import static com.kishlaly.ta.utils.Dates.getBarTimeInMyZone;
import static com.kishlaly.ta.utils.Dates.getTimeInExchangeZone;


public class EMA extends EntityWithDate {

    private Double value;

    public EMA(final Long timestamp, final Double value) {
        super(timestamp);
        this.value = value;
        this.nativeDate = getTimeInExchangeZone(timestamp, exchangeTimezome).toString();
        this.myDate = getBarTimeInMyZone(timestamp, exchangeTimezome).toString();
    }

    public Long getTimestamp() {
        return this.timestamp;
    }

    public Double getValue() {
        return this.value;
    }

    public void setTimestamp(final Long timestamp) {
        this.timestamp = timestamp;
    }

    public String getNativeDate() {
        return this.nativeDate;
    }

    public void setNativeDate(final String nativeDate) {
        this.nativeDate = nativeDate;
    }

    public String getMyDate() {
        return this.myDate;
    }

    public void setMyDate(final String myDate) {
        this.myDate = myDate;
    }

    public void setValue(final Double value) {
        this.value = value;
    }

    public boolean valuesPresent() {
        return !Double.isNaN(value);
    }
}

File: src/main/kotlin/com/kishlaly/ta/model/indicators/EMA.kt
package com.kishlaly.ta.model.indicators

import com.kishlaly.ta.model.AbstractModel

class EMA(timestamp: Long, val value: Double) : AbstractModel(timestamp) {

    override fun valuesPresent() = value != Double.NaN

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as EMA

        if (value != other.value) return false

        return true
    }

    override fun hashCode(): Int {
        return value.hashCode()
    }

}